rosboard/html/js/viewers/WaypointPointCloudViewer.js

```diff
@@ -214,19 +234,30 @@ class WaypointPointCloudViewer extends Viewer {
             points[3 * i + 2] = (points_view.getUint16(offset + 4, true) / 65535) * zrange + zmin;
         }
 
+        // Add color to the points based on elevation
         const colors = [];
-        for (let i = 0; i < points.length; i+=3) {
+        for (let i = 0; i < points.length; i += 3) {
             const y = points[i + 2];
-            const hue = (((y - 0) * (1 - 0)) /  (0 + 3)) + 0;
+            const hue = (((y - 0) * (1 - 0)) / (0 + 3)) + 0;
             let color = new THREE.Color();
             color.setHSL(hue, 1.0, 0.5);
             colors.push(color.r, color.g, color.b);
         }
-        
+
         this.pointsBuffer.setAttribute('color', new THREE.Float32BufferAttribute(colors, 3));
         this.pointsBuffer.setAttribute('position', new THREE.BufferAttribute(points, 3));
         this.pointsBuffer.rotateX(-Math.PI / 2);
     }
+
+    getBotToGoalAngle(botName) {
+        if (this.bots[botName].position.x == null) {
+            return null;
+        }
+
+        const dx = this.bots[botName].waypoint.position.x - this.bots[botName].position.x;
```

Comment:
Should we check to verify waypoint is there?

Reply:
Good call, done